### PR TITLE
UIQM-5 Reorder field/row in quickMARC

### PR DESCRIPTION
**Pull request by uladislausamets**
## Purpose
https://issues.folio.org/browse/UIQM-5

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js [hunk 1]
```diff
@@ -55,6 +73,30 @@ const QuickMarcEditorRows = ({ name, fields, mutators: { addRecord, deleteRecord
               data-test-quick-marc-editor-row
               data-testid="quick-marc-editorid"
             >
+              <div className={styles.quickMarcEditorMovingRow}>
+                {
+                  !withMoveUpRowAction && (
+                    <IconButton
+                      data-test-move-up-row
+                      data-index={idx}
+                      data-nearest-index={idx - 1}
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js:82]: do you really need this attribute? you can get in function -1 or +1 based on direction

File: src/QuickMarcEditor/QuickMarcEditorRows/utils.js [hunk 2]
```diff
@@ -21,8 +21,14 @@ const ADD_EXCEPTION_ROWS = [LEADER_TAG, '001', '003', '005', '008'];
 
 export const hasAddException = recordRow => ADD_EXCEPTION_ROWS.includes(recordRow.tag);
 
-const DELETE_EXCEPTION_ROWS = [LEADER_TAG, '001', '005', '008'];
+const DELETE_MOVE_EXCEPTION_ROWS = [LEADER_TAG, '001', '005', '008'];
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/utils.js:24]: imo it's better to create new array

File: src/QuickMarcEditor/QuickMarcEditorRows/utils.js [hunk 3]
```diff
@@ -21,8 +21,14 @@ const ADD_EXCEPTION_ROWS = [LEADER_TAG, '001', '003', '005', '008'];
...
 
 export const hasDeleteException = recordRow => (
-  DELETE_EXCEPTION_ROWS.includes(recordRow.tag) || isLastRecord(recordRow)
+  DELETE_MOVE_EXCEPTION_ROWS.includes(recordRow.tag) || isLastRecord(recordRow)
+);
+
+export const hasMoveException = (recordRow, nearestRecordRow) => (
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/utils.js:30]: mb `sibling` is better name?

File: src/QuickMarcEditor/QuickMarcEditorRows/utils.test.js [hunk 4]
```diff
@@ -58,4 +58,15 @@ describe('QuickMarcEditorRows utils', () => {
...
     });
   });
+
+  describe('hasMoveException', () => {
+    it('should be true for exeptional row', () => {
+      expect(utils.hasMoveException({ tag: LEADER_TAG })).toBeTruthy();
+      expect(utils.hasMoveException({ tag: '001' })).toBeTruthy();
+    });
+
+    it('should be false for exeptional row', () => {
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/utils.test.js:68]: it's not exceptional case

File: test/bigtest/tests/QuickMarcEditor-test.js [hunk 5]
```diff
@@ -64,6 +72,17 @@ describe('Quick MARC editor', () => {
...
   });
 
+  describe('reorder records by clickin on move buttons', () => {
+    beforeEach(async function () {
+      await quickMarcEditor.moveRowDownButton.click();
+      await quickMarcEditor.moveRowUpButton.click();
+    });
+
+    it('records count should not change (including leader row)', () => {
+      expect(quickMarcEditor.editorRows().length).to.be.equal(records.length + 1);
```
Comment (NikitaSedyx) [inline test/bigtest/tests/QuickMarcEditor-test.js:82]: mb it's better to check order is changed?

File: src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js (hunk 1 above)
Comment (uladislausamets) [reply src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js:82]: @NikitaSedyx  In this case, I need 2 separate functions (to move up and move down). Do I have any better ways to know direction in one function?

Comment (NikitaSedyx) [reply src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js:82]: it's correct, 2 functions

File: src/QuickMarcEditor/QuickMarcEditor.js [hunk 6]
```diff
@@ -111,6 +112,11 @@ export default stripesFinalForm({
     deleteRecord: ([{ index }], state, tools) => {
       const records = deleteRecordByIndex(index, state);
 
+      tools.changeValue(state, 'records', () => records);
+    },
+    moveRecord: ([{ index, nearestIndex }], state, tools) => {
+      const records = reorderRecords(index, nearestIndex, state);
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditor.js:118]: nearestIndex -> indexToChange or smth like that?

File: src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js (hunk 1 above)
Comment (NikitaSedyx) [reply src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js:82]: it's ok to have 1 function, `nearest-index` doesn't say a lot

**Pull request merged**